backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -1153,6 +1186,54 @@ private class InteropTransformer(val context: Context, override val irFile: IrFi
         }
     }
 
+    override fun visitTypeOperator(expression: IrTypeOperatorCall): IrExpression {
```

Comment:
What is handling Obj-C type checks here required for?

Reply:
To let analyses see some of the underlying magic. For instance, for escape analysis it's not clear why `is` operator might force argument to escape.

Reply:
Previous implementation for Obj-C type checks was designed with avoiding code duplication in mind. See comments with reproducers for related bugs in new implementation.
So it might be easier to let analyses see the underlying magic.

Reply:
I thought that the more we can do in lowerings and not in codegen the better.

Reply:
It depends. In this particular case it might be true, but avoiding code duplication (to support type operand erasure, safe casts etc.) is important too. So, it would be great to achieve both.